File: TrabajosPrevios/Sesion4/HerenciaMultiple.cpp
#include <iostream> 

/*
El programa muestra como una misma clase puede 
heredar a varias clases.
*/

using namespace std;

class Mammal {
    public:
     Mammal() {
      cout << "Mammals can give direct birth." << endl;
    }
};
class WingedAnimal {
    public:
     WingedAnimal() {
      cout << "Winged animal can flap." << endl;
    }
};
class Bat: public Mammal, public WingedAnimal {};

int main() {
    Bat b1;
  return 0;
}

File: TrabajosPrevios/sesion12/herenciaMultiple.py
class Mammal:
    def mammal_info(self):
        print("Mammals can give direct birth.")
class WingedAnimal:
    def winged_animal_info(self):
        print("Winged animals can flap.")
class Bat(Mammal, WingedAnimal):# Hereda metodos de las dos clases
    pass
# Crea un objeto de la clase Bat
b1 = Bat()
b1.mammal_info()
b1.winged_animal_info()